### DISCUSSION TITLE
Can't connect via art-net or sACN

### DISCUSSION THREAD

**Discussion by Marianskiii** [forkineye/ESPixelStick#771] [Q&A]
Hello there,

I've uploaded the latest 4 build (tried also 3.2) on esp32 with max485 to dmx board, but whatever i do cannot connect via art-net of sACN.
Tried with obsidian onyx, qlc+, madmapper and resolume, none of them can send to the esp32.
For example when i try ddp from LedFX to esp32 it's working without a problem.

Any ideas?

**Comment by MartinMueller2003** [comment]
Try sACN View (free tool) and please post your device configuration.

**Reply by Marianskiii** [reply]
Here's some screenshots:
![home](https://github.com/forkineye/ESPixelStick/assets/174797679/79bdd7e6-ec73-4d6c-b971-2d525929965f)
![device setup](https://github.com/forkineye/ESPixelStick/assets/174797679/0a1566de-cbe9-408e-b25d-aa33e59fe95d)
![sACN view](https://github.com/forkineye/ESPixelStick/assets/174797679/d866b8ed-cd3e-4d0d-8e33-b693176ad375)

Setup is - Onyx for controlling dmx fixtures that outputs sACN or art-net to the espixelstick via home network. PC true wire to router -> wifi to esp32 -> rs-485 to xlr.

As shown on screenshots, onyx is outputting sACN, but esp32 cannot receive it for some reason.

I have other ethernet to dmx controllers and they work fine in that network.  

Here's a log when i chose art-net for input:

[178966][E][vfs_api.cpp:104] open(): /littlefs/www/index.html does not exist, no permits for creation
23:49:22: [FileMgr] Configuration File: '/output_config.json' saved 1041 bytes.
23:49:36: [FileMgr] Configuration File: '/output_config.json' saved 1436 bytes.
23:49:36: [FileMgr] Configuration File: '/output_config.json' saved 1436 bytes.
23:49:36: [FileMgr] Configuration File: '/output_config.json' saved 1436 bytes.
23:49:37: [FileMgr] Configuration File: '/input_config.json' saved 1043 bytes.
23:49:37: [FileMgr] Configuration File: '/input_config.json' saved 948 bytes.
23:49:37: [FileMgr] Configuration File: '/input_config.json' loaded.
23:49:37: [InputMgr] Shutting Down 'E1.31' on Input: 0
23:49:37: [InputMgr] Starting Artnet for channel '0'.
23:49:37: [Artnet] Subscribed to broadcast
23:49:37: [Artnet] Listening for 64 channels from Universe 1 to 1
23:49:37: [Artnet] Listening for 64 channels from Universe 1 to 1
23:49:37: [FileMgr] Configuration File: '/config.json' saved 374 bytes.
23:49:37: [FileMgr] Configuration File: '/config.json' loaded.
23:49:37: [FileMgr] Configuration File: '/output_config.json' saved 1436 bytes.
23:49:37: [FileMgr] Configuration File: '/output_config.json' saved 1436 bytes.
23:49:37: [FileMgr] Configuration File: '/output_config.json' saved 1436 bytes.
23:49:37: [FileMgr] Configuration File: '/output_config.json' saved 478 bytes.
23:49:37: [FileMgr] Configuration File: '/output_config.json' loaded.
23:49:38: [Artnet] Listening for 64 channels from Universe 1 to 1
23:49:38: [   ESP] ***-----Internal Reboot Requested. Rebooting Now
23:49:39: [WiFiDrv] WiFi Lost the connection to the AP
23:49:39: [WiFiDrv] WiFi Entering State: Connection Failed
23:49:39: ets Jul 29 2019 12:21:46
23:49:39: 
23:49:39: rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
23:49:39: configsip: 0, SPIWP:0xee
23:49:39: clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
23:49:39: mode:DIO, clock div:2
23:49:39: load:0x3fff0030,len:1184
23:49:39: load:0x40078000,len:12784
23:49:39: load:0x40080400,len:3032
23:49:39: entry 0x400805e4
23:49:39: E (330) psram: PSRAM ID read error: 0xffffffff
23:49:40: 
23:49:40: [   ESP] ESPixelStick v4.0-beta5 (Dec  1 2023 - 17:11:40)
23:49:40: [   ESP] v4.4.1-1-gb8050b365e
23:49:40: [FileMgr] Flash file system initialized. Used = 217088 out of 327680
23:49:40: [FileMgr] Listing directory: /
23:49:40: [FileMgr] 'UpdRecipe.json': 	'784'
23:49:40: [FileMgr] 'admininfo.json': 	'160'
23:49:40: [FileMgr] 'config.json': 	'374'
23:49:40: [FileMgr] 'input_config.json': 	'1991'
23:49:40: [FileMgr] 'output_config.json': 	'10135'
23:49:40: [FileMgr] Listing directory: //www
23:49:40: [FileMgr] 'alexa.html.gz': 	'147'
23:49:40: [FileMgr] 'apa102.html.gz': 	'1494'
23:49:40: [FileMgr] 'artnet.html.gz': 	'384'
23:49:41: [FileMgr] 'ddp.html.gz': 	'145'
23:49:41: [FileMgr] 'dmx.html.gz': 	'454'
23:49:41: [FileMgr] 'e1_31.html.gz': 	'462'
23:49:41: [FileMgr] 'effects.html.gz': 	'971'
23:49:41: [FileMgr] 'esps.css.gz': 	'13393'
23:49:41: [FileMgr] 'esps.js.gz': 	'60981'
23:49:41: [FileMgr] 'favicon.ico': 	'6450'
23:49:41: [FileMgr] 'fpp_remote.html.gz': 	'346'
23:49:41: [FileMgr] 'gece.html.gz': 	'499'
23:49:41: [FileMgr] 'gs8208.html.gz': 	'1575'
23:49:41: [FileMgr] 'index.html.gz': 	'4937'
23:49:41: [FileMgr] 'mqtt.html.gz': 	'648'
23:49:41: [FileMgr] 'relay.html.gz': 	'505'
23:49:41: [FileMgr] 'renard.html.gz': 	'528'
23:49:41: [FileMgr] 'serial.html.gz': 	'619'
23:49:41: [FileMgr] 'servo_pca9685.html.gz': 	'519'
23:49:41: [FileMgr] 'tls3001.html.gz': 	'1495'
23:49:41: [FileMgr] 'tm1814.html.gz': 	'1610'
23:49:41: [FileMgr] 'ucs1903.html.gz': 	'1576'
23:49:41: [FileMgr] 'ucs8903.html.gz': 	'1578'
23:49:41: [FileMgr] 'ws2801.html.gz': 	'1492'
23:49:41: [FileMgr] 'ws2811.html.gz': 	'1577'
23:49:41: [  4019][E][sd_diskio.cpp:802] sdcard_mount(): f_mount failed: (3) The physical drive cannot work
23:49:44: [  4519][E][sd_diskio.cpp:126] sdSelectCard(): Select Failed
23:49:44: [FileMgr] No SD card installed
23:49:44: [FileMgr] Configuration File: '/config.json' loaded.
23:49:44: [FileMgr] Configuration File: '/input_config.json' loaded.
23:49:44: [FileMgr] Configuration File: '/output_config.json' loaded.
23:49:45: [FileMgr] Configuration File: '/admininfo.json' saved 160 bytes.
23:49:46: [FileMgr] Configuration File: '/config.json' saved 360 bytes.
23:49:46: [WiFiDrv] WiFi Entering State: Connecting Using Config Credentials
23:49:46: [WiFiDrv] Using DHCP
23:49:46: [WiFiDrv] Connecting to 'vj' as test123345
23:49:47: [WiFiDrv] WiFi Entering State: Connected To AP
23:49:48: [WiFiDrv] Connected with IP: 192.168.1.29
23:49:48: [Artnet] Subscribed to broadcast
23:49:48: [Artnet] Listening for 64 channels from Universe 1 to 1
23:49:48: [WebMgr] Web server listening on port 80
23:49:48: [  FPPD] FPPDiscovery subscribed to broadcast messages on port: 32320
23:49:48: [  FPPD] FPPDiscovery subscribed to multicast: 239.70.80.80:32320
23:49:48: POLL from 192.168.1.2 broadcast addr: 10.0.1.255
23:49:57: POLL from 192.168.1.2 broadcast addr: 10.0.1.255
23:50:17: [ 39637][E][vfs_api.cpp:104] open(): /littlefs/www/index.html does not exist, no permits for creation
23:50:19: POLL from 192.168.1.2 broadcast addr: 10.0.1.255
23:50:28: POLL from 192.168.1.2 broadcast addr: 10.0.1.255
23:50:38:

**Comment by MartinMueller2003** [comment]
sACN is set to send to 192.168.1.2
Your device is set to 192.168.1.23
You packets are not being sent to the controller

**Reply by Marianskiii** [reply]
Well, the source of sACN is 192.168.1.2, but i can't specify the end point of it in any of the software i use.
For the art-net on other hand - can specify the end point (esp32) and with wled it's just works. Here's an example:
![wled art-net3](https://github.com/forkineye/ESPixelStick/assets/174797679/f3e19026-cc76-4f55-a3ed-cbc880c20d98)

**Reply by MartinMueller2003** [reply]
In this picture you are showing the DESTINATION of the artnet data. In sACN View you need to Specify the destination, not the source. The source address will be that of the computer running sACN view.

**Comment by MartinMueller2003** [comment]
Try this screen:

![image](https://github.com/forkineye/ESPixelStick/assets/8608503/7a3f7579-5a2a-487a-9320-1000618f6ce0)

**Reply by Marianskiii** [reply]
Thank you very much for the help. i really appreciate it.

Had several issues with my setup;
first of all sACN view was on the same computer as the source signal. duh.
Now i see the sACN stream into esp32. Problem was with my ISP router (don't know what exactly), but when i change it with generic, clean tp-link sACN now works.
That's nice, but i still can't manage to make art-net to work.
Did try with the override to esp32 ip and without in onyx, but it won't receive any data.
My other DIY (no model) ethernet-dmx controller works in the same setup.

**Comment by MartinMueller2003** [comment]
I have been trying to find a free artnet data source that I can use to test. Once I find one I can see if there is something odd going on. I have tried three so far and they all failed to load on my pc (ok  magic-q loaded but is so complicated I gave up on it)

**Reply by Marianskiii** [reply]
Obsidian onyx it's free for test. https://support.obsidiancontrol.com/Content/Support/Downloads.htm
On the free version there is one ethernet universe free for testing purposes. 

Also i'm using Protol to capture art-net data and it's going out of the source pc to other with the utility, but no luck getting it in esp32.

**Comment by MartinMueller2003** [comment]
So far I have gotten onyx to send out the broadcast polls and they are being properly received on the ESP32. I did note a bug in the library. It is sending a message on the serial port every time a poll is received. That wont work on ESPs with 8 ports. I had to modify the library to send the message via the input driver which has the code to not transmit console data when the console output is being used for pixel data.

No idea how to get this thing to play the demo sequences that the SW came with. I suspect it is going to work fine once I find the play button.

**Comment by MartinMueller2003** [comment]
I just uploaded a set of changes that allowed support for universe zero in artnet. Onyx is now broadcasting data to the ESP32 device. Because the artnet application is a polled application, it is losing almost 40% of the PDUs. It will need to be converted into an async application so that the poll from the main loop is not needed. At any rate, it is working at this time. You can download my latest CI build or use the most recent CI build on the web flasher.

**Comment by MartinMueller2003** [comment]
Converted to Async UDP and the number of lost PDUs dropped 10%. Still losing a lot but I am not going to be able to do anything about it.

**Comment by Marianskiii** [comment]
Thanks again.
Now art-net is working fine, even with that 10% packet errors i cannot see any visible artifacts.
In that matter sACN, at least on my setup on generic esp32 without external antenna is having much more errors and it's laggy as hell.
I'm waiting the new batch with esp32's with external antenna and will post how it goes.
